packages/common/src/hooks/useHarmonyField.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  FieldHelperProps,
+  FieldHookConfig,
+  FieldMetaProps,
+  FieldInputProps as FormikFieldInputProps,
+  useFormikContext,
+  useField as useFormikField
+} from 'formik'
+import { useDebouncedCallback } from './useDebouncedCallback'
+import { useCallback, useState } from 'react'
+
+/** Extracts the event target's value from the event for both React and ReactNative. */
+const getEventTargetValue = (
+  e: React.ChangeEvent<any> | React.FocusEvent<any>
+) => e.target?.value ?? e.currentTarget?.value ?? (e as any).nativeEvent?.text
+
+export type UseHarmonyFieldProps<Value> = FieldHookConfig<Value> & {
+  /** Function to transform the input value on change, eg. a function to trim whitespace */
+  transformValueOnChange?: (value: string) => string
+  /** Function to transform the input value on blur, eg. a function to trim whitespace */
+  transformValueOnBlur?: (value: string) => string
+  /**
+   * Delays onChange validation errors from showing. Useful for improving the
```

Comment:
this is probably not the highest priority (I cant even think of an exact scenario for it); but I would expect this debounce prop to behave the same for onBlur validations as well

Also side note; would this occur after transforms have occurred?

Reply:
I actually initially had it debounce validation for onBlur too anyway but I figure it might make sense to not delay that one unnecessarily.. but maybe this is causing a race condition 🤔 good callout

But yeah the debouncing is only for the validation, so yeah it's after the other transforms.